File: Data Structures & Algorithms/remove-element/submission-3.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        # j = 0
        # n = len(nums)
        # s = []
        
        # while j < n:
        #     if nums[j] != val:
        #         s.append(nums[j]) 
        #     j += 1
        
        # for i in range(len(s)):
        #     nums[i] = s[i]

        # return len(s)

        i = 0
        n = len(nums)

        while i < n:
            if nums[i] == val:
                n -= 1
                nums[i] = nums[n]
            else:
                i += 1
        return n 

File: Data Structures & Algorithms/partition-to-k-equal-sum-subsets/submission-3.go
func canPartitionKSubsets(nums []int, k int) bool {
    sum := 0
    
    for _, number := range nums {
        sum += number
    }

	if sum % k != 0 {
		return false
	}

	target := sum / k
    sides := make([]int, k)

	if nums[0] > target {
        return false 
    }

    sort.Sort(sort.Reverse(sort.IntSlice(nums)))

	var backtrack func(i int) bool
	backtrack = func(i int) bool {
        if i == len(nums) {
            return true
        }

        for j := 0; j < k; j++ {
            if sides[j] + nums[i] > target {
                continue
            }

            sides[j] +=  nums[i]

            if backtrack(i + 1) {
                return true
            }

            sides[j] -=  nums[i]

            if sides[j] == 0 {
                break
            }
        }

		return false
	}

	return backtrack(0)
}
